runtime/src/main/java/io/quarkiverse/githubapp/runtime/config/CheckedConfigProvider.java

```diff
@@ -0,0 +1,131 @@
+package io.quarkiverse.githubapp.runtime.config;
+
+import java.security.PrivateKey;
+import java.util.Optional;
+import java.util.Set;
+import java.util.TreeSet;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.jboss.logging.Logger;
+
+import io.quarkiverse.githubapp.ConfigFile;
+import io.quarkiverse.githubapp.GitHubEvent;
+import io.quarkiverse.githubapp.runtime.config.GitHubAppRuntimeConfig.Debug;
+import io.quarkus.runtime.LaunchMode;
+import io.quarkus.runtime.Startup;
+
+@Singleton
+@Startup
+public class CheckedConfigProvider {
+
+    private static final Logger LOG = Logger.getLogger(GitHubEvent.class.getPackageName());
+
+    private final GitHubAppRuntimeConfig gitHubAppRuntimeConfig;
+
+    private final LaunchMode launchMode;
+
+    private final Set<String> missingPropertyKeys = new TreeSet<>();
+
+    @Inject
+    CheckedConfigProvider(GitHubAppRuntimeConfig gitHubAppRuntimeConfig, LaunchMode launchMode) {
+        this.gitHubAppRuntimeConfig = gitHubAppRuntimeConfig;
+        this.launchMode = launchMode;
+
+        if (gitHubAppRuntimeConfig.appId.isEmpty()) {
+            missingPropertyKeys.add("quarkus.github-app.app-id (QUARKUS_GITHUB_APP_APP_ID)");
+        }
+        if (gitHubAppRuntimeConfig.privateKey.isEmpty()) {
+            missingPropertyKeys.add("quarkus.github-app.private-key (QUARKUS_GITHUB_APP_PRIVATE_KEY)");
+        }
+        if (launchMode == LaunchMode.NORMAL && gitHubAppRuntimeConfig.webhookSecret.isEmpty()) {
+            missingPropertyKeys.add("quarkus.github-app.webhook-secret (QUARKUS_GITHUB_APP_WEBHOOK_SECRET)");
+        }
+
+        if (launchMode != LaunchMode.TEST) {
+            checkConfig();
+        }
+
+        if (gitHubAppRuntimeConfig.webhookSecret.isPresent() && launchMode.isDevOrTest()) {
+            LOG.info("Payload signature checking is disabled in dev and test modes.");
+        }
+    }
+
+    public String appId() {
+        if (launchMode == LaunchMode.TEST) {
+            checkConfig();
+        }
+        // this will never happen but let's make sure the code analysis is aware of it
+        if (gitHubAppRuntimeConfig.appId.isEmpty()) {
+            throw new IllegalStateException("quarkus.github-app.app-id may not be null");
+        }
+
+        return gitHubAppRuntimeConfig.appId.get();
+    }
+
+    public Optional<String> appName() {
+        return gitHubAppRuntimeConfig.appName;
+    }
+
+    public boolean readConfigFilesFromSourceRepository() {
+        return gitHubAppRuntimeConfig.readConfigFilesFromSourceRepository;
+    }
+
+    public PrivateKey privateKey() {
+        if (launchMode == LaunchMode.TEST) {
+            checkConfig();
+        }
+        // this will never happen but let's make sure the code analysis is aware of it
+        if (gitHubAppRuntimeConfig.privateKey.isEmpty()) {
+            throw new IllegalStateException("quarkus.github-app.private-key");
+        }
+
+        return gitHubAppRuntimeConfig.privateKey.get();
```

Comment:
```suggestion
        // The optional will never be empty; using orElseThrow instead of get to avoid IDE warnings.
        return gitHubAppRuntimeConfig.privateKey.orElseThrow();
```